mlir/lib/Dialect/Rock/Transforms/AffixTuningParameters.cpp

```diff
@@ -167,15 +167,13 @@ void AffixTuningParameters::affixBackwardWeightUtilityKernels(
     GemmContext gemmSize =
         GemmContext::fromConvolution(ConvOpType::BwdWeight, convDims);
 
-    PopulateParamsXDL populateParamsXDL;
+    auto gemmParams =
+        op->getAttrOfType<XdlopsGemmParamsAttr>(op.paramsAttrName());
     Optional<GemmContext> extraPadSizes =
-        calculatePaddingKernelSize(gemmSize, obtainConvDirection(op),
-                                   obtainConvDataType(op), populateParamsXDL);
+        calculatePadding(gemmParams.getKPerBlock(), gemmParams.getMPerBlock(),
+                         gemmParams.getNPerBlock(), gemmSize);
 
-    // For padding cases, gemmId must be 0.
-    if (extraPadSizes.hasValue()) {
-      assert(gemmId == 0);
```

Comment:
I had to remove this to make a case : mlir/test/rocmlir-driver/auto_e2e/conv2d_f16_wrw_perf_config.mlir (in this PR to work). I encountered this case in a tuning run on the weekly which is only triggered when a padded perf-config case was given.

Reply:
It is worth noting that this path was hardly taken before as it used to run calculatePaddingKernelSize that out right ignores perf-config as discussed before

Reply:
This assertion is there to assert that we haven't added the zeroing utility kernel when there's no padding.

So this means you may need to fix the detection if whether there'll be padding or not to account for perf configs in order to get this PR to work.

I'd put the assert back and fix the underlying problem.

Reply:
That does not make sense to me.
 
> This assertion is there to assert that we haven't added the zeroing utility kernel when there's no padding.

extraPadSizes.hasValue() reads as a scenario where there is padding as opposed to a scenario where there is no padding. if anything it is checking when there is padding, it seems to generate only a single kernel ? if not what does gemmId means here ?

Reply:
gemmId is the ID of the kernel.
In backwards weight, the cases are this:
1. There  is padding: there is only one kernel, the gemmId is 0
2. There is no padding, the datatype is FP32: There are two kernels, one that zeroes the result buffer and a second one that does the commputation. Their gemmIDs are 0 and 1, respectively
3. There is no padding, the datatype is FP16: There are three kernels, one that zeroes the workspace, one that computes the convolution to the workspace, and one that copies the workspace to the output with truncation

The number of kernels that the system is generating must match those cases, and the assertion ensures that

Reply:
(ok, yeah, I had it backwards: you need to only generate one kernel when there is padding)

The way I could see this assert tripping is that if you have a case where one of the heuristic parameter sets leads to a no-padding case (giving you a zeroing kernel) but the perf config leads to the padding case.

Therefore, the code to check if a backwards weight kernel needs padding (as used in Miir -> hasWorkspace()) needs to be modified to take the tuning parameters.

This might also involve finally writing the function that goes from a perf_config to a [Something]GemmParamsAttr

Reply:
Thanks for enlightenment of what is going in this part of the code...

I managed to get it working with assert.